### PR TITLE
Feature/wa 533 manage tokens vs

### PR DESCRIPTION
**Pull request by sche**
- AddTokenTableViewController with tests added
- minimum version update

### CODE REVIEW

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/ManageTokensTableViewController.swift [hunk 1]
```diff
@@ -42,7 +42,8 @@ class ManageTokensTableViewController: UITableViewController {
     }
 
     @objc private func addToken() {
-        // TODO
+        let controller = AddTokenTableViewController.create()
```
Comment (sche) [inline SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/ManageTokensTableViewController.swift:45]: will be done by flow coordinator

File: IdentityAccess/IdentityAccessImplementations/biometry/BiometricService.swift [hunk 2]
```diff
@@ -57,19 +57,15 @@ public final class BiometricService: BiometricAuthenticationService {
...
-            case .touchID: return .touchID
-            case .none:
-                ApplicationServiceRegistry.logger.error("Received unexpected biometry type: none",
-                                                        error: BiometricServiceError.unexpectedBiometryType)
-                return .none
-            }
-        } else {
-            return isAuthenticationAvailable ? .touchID : .none
+        guard isAuthenticationAvailable else { return .none }
+        // biometryType available from iOS 11.0
```
Comment (DmitryBespalov) [inline IdentityAccess/IdentityAccessImplementations/biometry/BiometricService.swift:61]: can be removed

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    private func token(for indexPath: IndexPath) -> TokenData {
+        return sectionedTokens[sectionTokensTitles[indexPath.section]]![indexPath.row]
+    }
+
+}
+
+extension AddTokenTableViewController: UISearchResultsUpdating {
+
+    func updateSearchResults(for searchController: UISearchController) {
+        guard let text = searchController.searchBar.text?.lowercased(), text != "" else {
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift:120]: !text.isEmpty

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            filteredTokens = tokens
+            return
+        }
+        filteredTokens = tokens.filter {
+            $0.code.lowercased().contains(text) || $0.name.lowercased().contains(text)
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift:125]: we should use `.localizedCaseInsensitiveContains(text)` for any user facing strings

**Review by DmitryBespalov — APPROVED** [review]
some small comments

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift (hunk 4 above)
Comment (sche) [reply SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift:125]: token code is not localisable.

File: IdentityAccess/IdentityAccessImplementations/biometry/BiometricService.swift (hunk 2 above)
Comment (sche) [reply IdentityAccess/IdentityAccessImplementations/biometry/BiometricService.swift:61]: ok

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift (hunk 3 above)
Comment (sche) [reply SafeUIKit/SafeAppUI/AppFlow/MainFlow/Tokens/ManageTokens/AddTokenTableViewController.swift:120]: ok

**Pull request merged**